### PR TITLE
Fix actions not being saved to databse on import from file

### PR DESCRIPTION
**Pull request by ClFeSc**
There was a person who wrote code that was never able to work.

### CODE REVIEW

**Review by Dassderdie — APPROVED** [review]
TBH - I don't  really understand this code - but because currently time is of the essence 👍

**Comment by ClFeSc** [comment]
@Dassderdie Do you have any questions about this? I think this was the reason why this is not yet merged.

**Comment by ClFeSc** [comment]
@Dassderdie I'd really like to get this merged sometime soon, as users will stumble upon this and be confused.

File: backend/src/exercise/exercise-wrapper.ts [hunk 1]
```diff
@@ -319,18 +319,21 @@ export class ExerciseWrapper extends NormalType<
...
         }
         this.validateInitialState();
-        this.restoreState();
+        this.restoreState(keepActions);
     }
 
-    private restoreState() {
+    /**
+     * @param keepActions This indicates whether to keep the actions that were applied while restoring in the array (when `true`) or to remove them (when `false` and when the database gets used)
+     */
```
Comment (Dassderdie) [inline backend/src/exercise/exercise-wrapper.ts:335]:
```suggestion
     * Recreates the {@link currentState} by applying all actions from {@link temporaryActionHistory} to the {@link initialState}
     * as well as adding actions to the end to gracefully mark the end of the previous exercise session.

     */
```

File: backend/src/exercise/exercise-wrapper.ts [hunk 2]
```diff
@@ -353,8 +356,8 @@ export class ExerciseWrapper extends NormalType<
         this.incrementIdGenerator.setCurrent(
             this.temporaryActionHistory.length
         );
-        // Remove all actions to not save them again (if database is active)
-        if (Config.useDb) {
+        // Remove all actions to not save them again (if database is active and this is intended behavior in this context)
+        if (Config.useDb && !keepActions) {
             this.temporaryActionHistory.splice(
```
Comment (Dassderdie) [inline backend/src/exercise/exercise-wrapper.ts:361]:
```suggestion
        if (Config.useDb && !keepActions) {
            // Remove all actions to not save them again in the database
            this.temporaryActionHistory.splice(
```

File: backend/src/exercise/exercise-wrapper.ts [hunk 3]
```diff
@@ -287,7 +287,7 @@ export class ExerciseWrapper extends NormalType<
                     )
             );
             pushAll(exercise.temporaryActionHistory, actions);
-            exercise.restore();
+            exercise.restore(true);
```
Comment (Dassderdie) [inline backend/src/exercise/exercise-wrapper.ts:290]:
```suggestion
            // The actions haven't been saved in the database yet -> keep them
            exercise.restore(true);
```

File: backend/src/exercise/exercise-wrapper.ts [hunk 4]
```diff
@@ -434,7 +437,7 @@ export class ExerciseWrapper extends NormalType<
                 })
             );
             await Promise.all(
-                exercises.map(async (exercise) => exercise.restore())
+                exercises.map(async (exercise) => exercise.restore(false))
```
Comment (Dassderdie) [inline backend/src/exercise/exercise-wrapper.ts:440]:
```suggestion
                // The actions have already been saved in the database -> do not keep them
                exercises.map(async (exercise) => exercise.restore(false))
```

**Review by Dassderdie — APPROVED** [review]
If my comments are correct, I'm ok with this.

**Pull request merged**